stablehlo/tools/StablehloTranslateMain.cpp

```diff
@@ -56,49 +55,87 @@ llvm::Error evalCustomCallCheckEq(stablehlo::CustomCallOp op,
             : stablehlo::check::evalExpectAlmostEqOp(actualResult,
                                                      expectedResult);
   if (status)
-    return stablehlo::invalidArgument(
-        "Error evaluating function: %s. \n\tCheck almost_eq failed: "
-        "%s",
-        funcName.data(), toString(std::move(status)).c_str());
+    scope.add(op.getResults(), makeBooleanTensor(op->getContext(), false));
+  else
+    scope.add(op.getResults(), makeBooleanTensor(op->getContext(), true));
 
-  scope.add(op.getResults(), makeBooleanTensor(op->getContext(), true));
-  return llvm::Error::success();
+  return status;
 }
 
 llvm::Error interpreterFallback(Operation &op, stablehlo::Scope &scope,
                                 llvm::StringRef funcName) {
-  // TODO: Write a pass to inline this custom call into a check op.
+  // custom_call(...) implementation
   if (auto customCall = dyn_cast<stablehlo::CustomCallOp>(op)) {
-    if (customCall.getCallTargetName() == "check.eq")
-      return evalCustomCallCheckEq(customCall, scope, funcName);
+    if (customCall.getCallTargetName() == "check.eq") {
+      auto status = evalCustomCallCheckEq(customCall, scope);
+      if (status)
+        return stablehlo::invalidArgument(
+            "Error evaluating function: %s. \n\tCheck eq failed: %s",
+            funcName.data(), toString(std::move(status)).c_str());
+      return llvm::Error::success();
```

Comment:
Can we create a helper for this block of code?

```C++
if (status)
    return stablehlo::invalidArgument(
        "Error evaluating function: %s. \n\tCheck eq failed: %s",
        funcName.data(), toString(std::move(status)).c_str());
  return llvm::Error::success();
```

It seems to be repeated 5 times in this file, which is a bit too much I think.

```C++
llvm::Error wrapStatus(llvm::Error status, llvm::StringRef funcName,
                       llvm::StringRef fallbackName) {
  if (status)
    return stablehlo::invalidArgument(
        "Error evaluating function: %s. \n\tFallback for %s failed: %s",
        funcName.data(), fallbackName.data(),
        toString(std::move(status)).c_str());
  return llvm::Error::success();
}
```

Reply:
+1